packages/react-native/scripts/codegen/__tests__/generate-specs-cli-executor-test.js

```diff
@@ -34,25 +34,21 @@ describe('generateSpec', () => {
         expect(encoding).toBe('utf-8');
         return fixtures.schemaText;
       },
-    }));
-
-    let mkdirpSyncInvoked = 0;
-    jest.mock('mkdirp', () => ({
-      sync: folder => {
-        if (mkdirpSyncInvoked === 0) {
-          expect(folder).toBe(outputDirectory);
+      mkdirSync: jest.fn((folder, options) => {
+        if (folder === outputDirectory) {
+          expect(options).toEqual({ recursive: true });
         }
```

Comment:
This test does not verify that the first call was for value in `outputDirectory`.

It attempts to verify that if any call included value from `outputDirectory`, it contained options set to `{ recursive: true }` which isn't the same.

Reply:
I think the testing should be similar to how it's done in `scripts/__tests__/scm-utils-test.js`

Create mock function using `jest.fn()`, and validate it's Nth calls using https://jestjs.io/docs/expect#tohavebeennthcalledwithnthcall-arg1-arg2-

Reply:
Thanks for the feedback @trivikr 
I've updated the test to ensure it explicitly verifies that mkdirSync is called exactly once with outputDirectory and the correct options { recursive: true }. Since the function logic only requires one directory creation, I removed unnecessary checks for multiple invocations.